### PR TITLE
[21387] Fix build issues and update CI infrastructure to detect them

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description
Due to an infrastructure issue with an internal action for Windows, the CI was not properly applying some CMake arguments while testing Fast DDS on Windows. Among others, the examples were not built, so CI passed even though.

This PR fixes the build issue, and tests the changes in the infrastructure to ensure it does not happen anymore.

It was detected also a build issue when security and log info were disabled. In this PR it has been introduced a CI job that builds Fast DDS without any additional arguments / flags, and the error fix has been introduced too.

Finally, building with warnings as error has been set also from the main CMakeLists.txt file.

Related eProsima-CI pr:
- https://github.com/eProsima/eProsima-CI/pull/116

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- **N/A** Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- ❌ Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- ❌ Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 
    
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- **N/A** If this is a critical bug fix, backports to the critical-only supported branches have been requested.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by MiguelCompany** [comment]
@JesusPoderoso This needs a rebase, and fixing an unused variable warning on EDP.cpp

**Review by MiguelCompany — APPROVED** [review]

File: src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp [hunk 1]
```diff
@@ -856,6 +856,7 @@ bool EDP::pairingWriter(
 
     BaseWriter* writer = BaseWriter::downcast(W);
     const GUID_t& writer_guid = writer->getGuid();
+    static_cast<void*>(writer_guid);  // Void cast to force usage if LOG_INFO and SECURITY are disabled
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp:859]:
```suggestion
    static_cast<void>(writer_guid);  // Void cast to force usage if LOG_INFO and SECURITY are disabled
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp (hunk 1 above)
Comment (JesusPoderoso) [reply src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp:859]: 🤦

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Review by MiguelCompany — APPROVED** [review]

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**